third_party/move/move-compiler-v2/tests/lints/stackless_bytecode_lints/needless_mutable_reference_warn.move

```diff
@@ -0,0 +1,183 @@
+module 0xc0ffee::m {
+    fun consume_mut(x: &mut u64) {
+        *x = 10;
+    }
+
+    fun consume_immut(_x: &u64) {}
+
+    fun test1_no_warn(x: u64) {
+        let y = &mut x;
+        *y = 10;
+    }
+
+    fun test1_warn(x: u64): u64 {
+        let y = &mut x;
+        *y
```

Comment:
Add a similar test case but with type S, which doesn't have copy.  This should result in a move, and not cause a warning.

Reply:
I'm glad to hear it has the compile error.  `*&mut x` would apparently not have a compile error.  How about that?  Both with and without `copy` ability.  One should cause a warning, the other a compile error; I guess you'd need a separate test for  the compile error case.  Though it won't have an error today due to the bug.

All your examples right now are u64 values, so having some struct values in examples would be useful.

Maybe even vectors.